sqlparser/sqlparser.go

```diff
@@ -61,33 +62,41 @@ func (p *Parser) parseShardColumnPlaceholderIndex(valExpr vtparser.Expr) int {
 	return 0
 }
 
+func (p *Parser) parseVal(val *vtparser.SQLVal, queryBase *QueryBase) error {
+	if val.Type != vtparser.ValArg {
+		id, err := strconv.Atoi(string(val.Val))
+		if err != nil {
+			return errors.WithStack(err)
+		}
+		queryBase.ShardKeyID = Identifier(id)
+		return nil
+	}
+
+	placeholderIndex := p.parseShardColumnPlaceholderIndex(val)
+	if placeholderIndex == 0 {
+		return errors.New("cannot parse shard_key column provided by query argument")
+	}
+	queryBase.ShardKeyIDPlaceholderIndex = placeholderIndex
+	if len(queryBase.Args) >= placeholderIndex {
+		arg := queryBase.Args[placeholderIndex-1]
+		switch argType := arg.(type) {
+		case int, int8, int32, int64:
+			queryBase.ShardKeyID = Identifier(argType.(int64))
+		case uint, uint8, uint32, uint64:
+			queryBase.ShardKeyID = Identifier(argType.(uint64))
```

Comment:
Are `int16` and `uint16` unnecessary?

Reply:
Oops...!! Thanks for specifying , I will fix it .